spark_ec2.py

```diff
@@ -162,11 +183,74 @@ def setup_external_libs(libs):
 from boto.ec2.blockdevicemapping import BlockDeviceMapping, BlockDeviceType, EBSBlockDeviceType
 from boto import ec2
 
+import yaml
 
 class UsageError(Exception):
     pass
 
 
+def process_conf_file(file_path, parser):
+    """
+    Load configuration file and extract arguments.
+    """
+    # Loading the configuration file.
+    configuration = {}
+    try:
+        with open(file_path) as configuration_file:
+            configuration = yaml.safe_load(configuration_file)
+    except Exception as e:
+        print("[!] An error occured when loading the config file:\n{}".format(e), file=stderr)
+        sys.exit(1)
+
+    unneeded_opts = ("version", "help")
+
+    # mapppin_conf() aims to avoid maintaining a manual mapping dictionary.
+    # Transforms --option-like-this to option_like_this
+    def mapping_conf(opt):
+        normal_opt = str(opt).split("/")[-1]
+        trans_opt = normal_opt.strip("-")
+        trans_opt = trans_opt.replace("-", "_")
+        return {trans_opt: normal_opt}
+
+    map_conf = {}
+    [map_conf.update(mapping_conf(opt)) for opt in parser.option_list]
+    [map_conf.pop(unneeded_opt, None) for unneeded_opt in unneeded_opts]
+
+    # Generating path to identity file (located in ~/.ssh/)
+    home_dir = os.path.expanduser('~')
```

Comment:
Do you think that keeping the same semantics as command line options in the configuration file is better ?

Like this:
``` yml
--configuration-file: path
--slave: 10
...
...
...
--region: us-west-1
--copy-aws-credentials: true
```

By keeping the same semantic, we don't need the mapping function anymore.